src/org/zaproxy/zap/spider/parser/SpiderHtmlFormParser.java

```diff
@@ -114,6 +119,9 @@ public boolean parseResource(HttpMessage message, Source source, int depth) {
 		List<Element> forms = source.getAllElements(HTMLElementName.FORM);
 
 		for (Element form : forms) {
+			for (Attribute att : form.getAttributes()){
+				envAttributes.put(att.getKey(), att.getValue());
```

Comment:
The `envAttributes` should be reset before processing each form (in the attached test you can notice that the attributes of the first form are passed to the second).